metrics.go

```diff
@@ -0,0 +1,119 @@
+package go2sky
+
+import (
+	"context"
+	"github.com/SkyAPM/go2sky/logger"
+	"github.com/shirou/gopsutil/v3/cpu"
+	"github.com/shirou/gopsutil/v3/mem"
+	"log"
+	"os"
+	"runtime"
+	"time"
+)
+
+const (
+	defaultInterval            = 15 * time.Second
+	defaultLogPrefix           = "go2sky-golang-metric"
+	InstanceGolangHeap         = "instance_golang_heap"
+	InstanceGolangStack        = "instance_golang_stack"
+	InstanceGolangGCTime       = "instance_golang_gc_time"
+	InstanceGolangGCCount      = "instance_golang_gc_count"
+	InstanceGolangThreadNum    = "instance_golang_thread_num"
+	InstanceGolangGoroutineNum = "instance_golang_goroutine_num"
+	InstanceGolangCPUUsedRate  = "instance_golang_cpu_used_rate"
+	InstanceGolangMemUsedRate  = "instance_golang_mem_used_rate"
+)
+
+type RunTimeMetric struct {
+	// the Unix time when metrics were collected
+	Time int64
+	// the bytes of allocated heap objects
+	HeapAlloc int64
+	// the bytes in stack spans.
+	StackInUse int64
+	// the number of completed GC cycles since instance started
+	GCCount int64
+	// the latest gc pause time(NS)
+	GCPauseTime int64
+	// the number of goroutines that currently exist
+	GoroutineNum int64
+	// the number of records in the thread creation profile
+	ThreadNum int64
+	// the cpu Used float64
+	CpuUsedRate float64
+	// the Percentage of RAM used by programs
+	MemUsedRate float64
+}
+
+type MetricCollector struct {
+	ctx      context.Context
+	reporter MetricsReporter
+	instance string
+	service  string
+	interval time.Duration
+	logger   logger.Log
+}
+
+func InitMetricCollector(reporter MetricsReporter, interval time.Duration, cancelCtx context.Context) {
+	if interval == 0 {
```

Comment:
When the user set interval to 0, disable runtime metrics, HBU？

Reply:
I think it's a good idea, I will do it

Reply:
It should be disabled if interval is 0 or negative.

Reply:
Ok, I had done this.